cartoframes/data/observatory/catalog/entity.py

```diff
@@ -147,9 +150,21 @@ def _get_credentials(self, credentials=None):
 
         return _credentials
 
+    def _get_remote_full_table_name(self, user_project, user_dataset, public_project):
```

Comment:
It's weird having this method and the `_download` one in `Entity`. I'd suggest to refactor to a `DownloadableEntity` class and if we want to keep using inheritance, then make it to extend `Entity` and `Dataset` and `Geography` inherit from that `DownloadableEntity`.

If we are not going to do it now, please, let's create a separate issue so it can be prioritized at some point.

Reply:
`_get_remote_full_table_name` is useful in any case because we are basically getting this info in different places but IMO the entity is responsible for that. I will do the same for `Variable` in the future if everybody agrees.

About `_download` method is a private one and it is called from `Catalog` and `Geography` entities. For me, a `DownloadableEntity` makes sense but don't you think we already have a lot of classes in the catalog?